### ISSUE TITLE
Reproduce results

### ISSUE THREAD

**Issue by wongmia** [zekarias-tilahun/deepred#2]
Hi Zekarias,

Thanks for sharing codes. I tried to reproduce the results of your paper, and I used the default parameters of this repository (ran 'bash train.sh wikipedia' and 'bash eval.sh wikipedia'). However, I cannot get the results same as your paper. For example, I got MRR:0.7412 Recall: 0.7504 for Wikipedia, and the results in your paper is 0.885 for MRR and 0.889 for Recall@10. Could you tell me which parameters or steps I did wrong? Thanks in advance.

**Comment by zekarias-tilahun** [comment]
Hi,

Thank you for pointing out the issue. Did you try the legacy version, under the `legacy` directory? That was the code base which is used to obtain the results reported in the paper. It had some bugs, which I have just fixed. You  should be able to reproduce them now.

As for the hyper-parameters, they are the ones reported in the paper.

**Comment by wongmia** [comment]
Thanks, Zekarias.

I have one question. In **deepred/legacy/deepred/data.py, line 298-299**. I found the code 
"      user_nh, user_nh_dt, user_mask = self.neighborhood_sampler(user, current_time, args.nbr_size)
        item_nh, item_nh_dt, item_mask = self.neighborhood_sampler(user, current_time, args.nbr_size)
"
The input of these two **self.neighborhood_sampler** function are both **user**. I was confused about it. Look forward to your reply. Thanks in advance.

**Comment by zekarias-tilahun** [comment]
Hi wongmia,

thanks again for catching this, i think this is a mistake due to copy and paste. Line 299 should be
"item_nh, item_nh_dt, item_mask = self.neighborhood_sampler(item, current_time, args.nbr_size)"

**Comment by wongmia** [comment]
Thanks, Zekarias.

I have one more question. In your code, you use a trick to discard some old embeddings for the purpose of GPU memory usage. However, in my opinion, if you only select several relative-new items into the candidate sets, it may be unfair to evaluate the performance. When I vary the ratio to discard old items, I got different evaluation metric scores. When I input all 125979 items of the train datasets and select them as the candidate items, I got MRR:0.8220 Recall: 0.8587 for the test dataset (take Wikipedia dataset as an example). I am confused about this part. Maybe I miss some important things. Look forward to your reply. Thanks in advance.

**Comment by zekarias-tilahun** [comment]
Hi @wongmia,

sorry for the late reply, I accidentally overlooked your latest comment. 

Regarding your question, this is due to the limitation in GPU memory. Nonetheless, at the time of publication, i have tried a range of values up to 100,000 and I didn't observe a significant difference in performance. However, feel free to try it with larger values, just as you did with 125979, and report that in your writing if you have observed a significant gap.
